### PR TITLE
Refs #12720 - Allow puppet CA cache to work for https proxies

### PR DESCRIPTION
**Pull request by tbrisker**
When dumping CA certificates to cache, the status object is also dumped.
Since the status object points to the API resource, Marshal will try
dumping the API resource as well. This will fail for https proxies as
the SSL certificates cannot be dumped. This prevents the dumping of the
api link which will be recreated once the object is loaded back from
cache.

### CODE REVIEW

**Comment by lzap** [comment]
Any chance we get some test coverage, it sneaked to the develop there.

**Comment by tbrisker** [comment]
@lzap I'm not quite sure how we can test this, as we would need to create an ssl connection to initialize the ProxyAPI object, and anyways that code does not run when in test: https://github.com/tbrisker/foreman/blob/12720/lib/proxy_api/resource.rb#L12

File: app/services/proxy_status/base.rb [hunk 1]
```diff
@@ -18,6 +18,15 @@ def self.humanized_name
       'Base'
     end
 
+    #Allow dumping status objects to the cache without dumping the api endpoint - so SSL certs won't break cache
```
Comment (lzap) [inline app/services/proxy_status/base.rb]: LGTM except the reasoning here, can you reword it to something more clear (SSL cert is too broad term). I'd go with

```
# avoid marshalling @api
```

or similar.

Comment (tbrisker) [inline app/services/proxy_status/base.rb]: reworded.

**Comment by lzap** [comment]
Will test and merge tomorrow!

**Comment by lzap** [comment]
Merged as 2f84e724e28454d4fd9815dc32091048b01b94a1, thank you!
